core/src/main/scala/dagr/core/tasksystem/EitherTask.scala

```diff
@@ -23,24 +23,31 @@
  */
 package dagr.core.tasksystem
 
-/** Helps determine which task to perform for an [[EitherTask]] */
-object EitherTaskFlag extends Enumeration {
-  type EitherTaskFlag = Value
-  val LEFT, RIGHT = Value
-}
-
 object EitherTask {
+  sealed trait Choice
+  object Left  extends Choice
+  object Right extends Choice
+
+  /**
+    * Creates an [[EitherTask]] that wraps `choice` into a function that will be evaluated when the
+    * [[EitherTask]] is evaluated at execution time.
```

Comment:
For a task that is not a `UnitTask`, how would you describe the lifecycle?  In my mind, the invocation of `getTasks` _is_ execution time for non-unit tasks like `Either` or `Pipeline`, because that is when their work is done.  

I can change this specific doc to something like `will be evaluated lazily when all dependencies have been met and a choice must be made`, but I'm unclear on why you think the existing wording is wrong.